finagle/h2/src/main/scala/com/twitter/finagle/buoyant/h2/netty4/Netty4H2Transporter.scala

```diff
@@ -35,33 +33,30 @@ object Netty4H2Transporter {
     )
 
     val pipelineInit: ChannelPipeline => Unit =
-      params[TransportSecurity].config match {
-        case TransportSecurity.Insecure =>
-          params[param.ClientPriorKnowledge] match {
-            case param.ClientPriorKnowledge(false) =>
-              // TODO support h1 upgrades
-              throw new IllegalArgumentException("client prior knowledge must be enabled")
 
-            case param.ClientPriorKnowledge(true) =>
-              // Prior Knowledge: ensure messages are buffered until
-              // handshake completes.
-              p => {
-                p.addLast(AnyToHeapInboundHandler)
-                p.addLast(framer)
-                p.addLast(new BufferingConnectDelay); ()
-              }
-          }
-
-        case TransportSecurity.Secure() =>
-          // Netty4Transporter has already installed `ssl` and
-          // `sslConnect` channel handlers. The Netty4ClientTls handler
-          // replaces these handlers with the `tls` and `tlsConnect`
-          // handlers, which are configured to advertise h2 support.
-          p => {
-            p.addLast(Netty4ClientTls.handler(params))
+      if (params[Transport.ClientSsl].e.isDefined) {
+        // secure
```

Comment:
What does this actually mean?  Perhaps a clearer comment like `// TLS is disabled.`

But I'm slightly mystified as to how `ClientSsl.e.isDefined` means that tls should be disabled... Are you sure that this shouldn't be `isEmpty`?